azure-auth-helper/src/main/java/com/microsoft/azure/auth/configuration/AuthConfiguration.java

```diff
@@ -89,4 +100,46 @@ public void setHttpProxyPort(String httpProxyPort) {
         this.httpProxyPort = httpProxyPort;
     }
 
+    public List<ConfigurationProblem> validate() {
+        final List<ConfigurationProblem> results = new ArrayList<>();
+        final String errorMessagePostfix = StringUtils.isNotBlank(serverId) ? "server: " + serverId + " in settings.xml." : "<auth> configuration.";
```

Comment:
return when `serverId!=null`? Or the prompt will be confusing if there are server id and wrong auth configuration

Reply:
The <serverId> check should be done out of this class, when serverId is null, the caller must load a new AuthConfiguration instance and perform validation against the new AuthConfiguration instance, in AuthConfiguration  class, there is no maven related variables, so it is not possible to validate the property like 'serverId'